src/SMatrix.jl

```diff
@@ -29,10 +29,6 @@ immutable SMatrix{S1, S2, T, L} <: StaticMatrix{T}
 end
 
 @generated function check_smatrix_params{S1,S2,L}(::Type{Val{S1}}, ::Type{Val{S2}}, T, ::Type{Val{L}})
-    if !(T <: DataType) # I think the way types are handled in generated fnctions might have changed in 0.5?
```

Comment:
Probably changing `T <: DataType` to `T isa Type` here work (on v0.6, otherwise try `isa(T, Type)`). Could you check if this works for you?